pkg/kubectl/cmd/context.go

```diff
@@ -0,0 +1,48 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package cmd
+
+import (
+	"io"
+)
+
+type CmdContext struct {
+	In            io.Reader
+	Out           io.Writer
+	ErrOut        io.Writer
+	BaseCmd       string
+	CmdName       string
+	FullName      string
+	CmdDetailsURL string
+	SubCmdNames   []string
```

Comment:
> Or we actually can use the CmdName for setting the value create quota and forget about the SubCmdNames... I am very open on that

I think the name becomes `ctx.BaseCmd + " " + ctx.CmdName`.  Using a `ctx.SubCommand("next-name")` will be able to set the new `BaseCmd` to existing `BaseCmd + " " + CmdName` and set the new `CmdName`.